packages/devtools_app/lib/src/flutter/app.dart

```diff
@@ -188,23 +187,33 @@ class DevToolsAppState extends State<DevToolsApp> {
   }
 }
 
-class DevToolsScreen {
-  const DevToolsScreen({
-    @required this.screen,
-    this.controllerProvider,
+class DevToolsScreen<T> {
+  const DevToolsScreen(
+    this.screen, {
+    this.createController,
```

Comment:
nit: might be worthwhile to make this required and force users to pass in null if they don't want a controller just as a signal that this is something users should think about.